Libraries/Opc.Ua.Security.Certificates/X509Certificate/X509PfxUtils.cs

```diff
@@ -0,0 +1,190 @@
+/* Copyright (c) 1996-2020 The OPC Foundation. All rights reserved.
+   The source code in this file is covered under a dual-license scenario:
+     - RCL: for OPC Foundation members in good-standing
+     - GPL V2: everybody else
+   RCL license terms accompanied with this source code. See http://opcfoundation.org/License/RCL/1.00/
+   GNU General Public License as published by the Free Software Foundation;
+   version 2 of the License are accompanied with this source code. See http://opcfoundation.org/License/GPLv2
+   This source code is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+*/
+
+
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Opc.Ua.Security.Certificates
+{
+    /// <summary>
+    /// Utilities to create a Pfx.
+    /// </summary>
+    public static class X509PfxUtils
+    {
+        /// <summary>
+        /// The size of the block used to test a sign or encrypt operation.
+        /// </summary>
+        public const int TestBlockSize = 0x20;
+
+        /// <summary>
+        /// Return the key usage flags of a certificate.
+        /// </summary>
+        private static X509KeyUsageFlags GetKeyUsage(X509Certificate2 cert)
+        {
+            var allFlags = X509KeyUsageFlags.None;
+            foreach (X509KeyUsageExtension ext in cert.Extensions.OfType<X509KeyUsageExtension>())
+            {
+                allFlags |= ext.KeyUsages;
+            }
+            return allFlags;
+        }
+
+        /// <summary>
+        /// Verify RSA key pair of two certificates.
+        /// </summary>
+        public static bool VerifyRSAKeyPair(
+            X509Certificate2 certWithPublicKey,
+            X509Certificate2 certWithPrivateKey,
+            bool throwOnError = false)
+        {
+            bool result = false;
+            RSA rsaPrivateKey = null;
+            RSA rsaPublicKey = null;
+            try
+            {
+                // verify the public and private key match
+                rsaPrivateKey = certWithPrivateKey.GetRSAPrivateKey();
+#if NETSTANDARD2_1
+                // on .NET Core 3 
+                rsaPrivateKey.ExportParameters(true);
```

Comment:
You're about to use the key, why export it here and ignore the result?

Reply:
@bartonjs this is called by a try/catch to ensure the private keys are exportable. If not the cert is read with again with different exportable options. On .NET Core 3 I can end up in a situation where I have access to the private key for encrypt etc, but still not exportable. Thats why the extra check here to cause an exception if not.